sw/device/lib/dif/dif_edn.c

```diff
@@ -68,27 +68,75 @@ dif_result_t dif_edn_set_auto_mode(const dif_edn_t *edn,
   }
   DIF_RETURN_IF_ERROR(check_locked(edn));
 
-  uint32_t reg = mmio_region_read32(edn->base_addr, EDN_CTRL_REG_OFFSET);
-  reg = bitfield_field32_write(reg, EDN_CTRL_CMD_FIFO_RST_FIELD,
-                               kMultiBitBool4True);
-  mmio_region_write32(edn->base_addr, EDN_CTRL_REG_OFFSET, reg);
+  // Check that EDN is disabled.  If it is not disabled (e.g., through a call
+  // to `dif_edn_stop()`), we are not ready to change mode.
+  uint32_t ctrl_reg = mmio_region_read32(edn->base_addr, EDN_CTRL_REG_OFFSET);
+  const uint32_t edn_en =
+      bitfield_field32_read(ctrl_reg, EDN_CTRL_EDN_ENABLE_FIELD);
+  if ((enum multi_bit_bool)edn_en != kMultiBitBool4False) {
+    return kDifError;
```

Comment:
Would `kDifUnavailable` make more sense here?

Reply:
I considered that too but was unsure about the "[the IP] will finish in some amount of time" part of the definition of `kDifUnavailable`. My thinking was that unless some other code disables EDN, there is no guarantee that this condition will be fulfilled in some amount of time. So I chose `kDifError` instead.

Though since all of you have more experience with DIFs than I do, I have no problem with returning `kDifUnavailable` if you think that is the right thing to do.

Reply:
I see, `kDifError` is fine!